process_report/tests/unit_tests.py

```diff
@@ -329,6 +329,83 @@ def test_apply_credit_0002(self):
         self.assertEqual(800, credited_projects.loc[5, "Balance"])
 
 
+class TestBUSubsidy(TestCase):
+    def setUp(self):
+        data = {
+            "Invoice Month": [
+                "2024-03",
+                "2024-03",
+                "2024-03",
+                "2024-03",
+                "2024-03",
+                "2024-03",
+            ],
+            "Manager (PI)": ["PI1", "PI2", "PI3", "PI3", "PI4", "PI4"],
+            "Institution": [
+                "Boston University",
+                "Boston University",
+                "Harvard University",
+                "Harvard University",
+                "Boston University",
+                "Boston University",
+            ],
+            "Project - Allocation": [
+                "ProjectA-e6413",
+                "ProjectB-5t143t",
+                "ProjectC-t14334",
+                "ProjectD",
+                "ProjectE-test-r25135",
+                "ProjectF",
+            ],
+            "Cost": [1050, 100, 10000, 1000, 1050, 100],
+            "Credit": [1000, 100, 0, 0, 1000, 0],
+            "Balance": [50, 0, 10000, 1000, 50, 100],
+        }
+        self.dataframe = pandas.DataFrame(data)
+        output_file = tempfile.NamedTemporaryFile(delete=False, mode="w", suffix=".csv")
+        self.output_file = output_file.name
+        self.subsidy = 100
+
+    def test_apply_BU_subsidy(self):
+        process_report.export_BU_only(self.dataframe, self.output_file, self.subsidy)
+        output_df = pandas.read_csv(self.output_file)
+
+        self.assertTrue(
+            set(
+                [
+                    process_report.INVOICE_DATE_FIELD,
+                    "Project",
+                    process_report.PI_FIELD,
+                    process_report.COST_FIELD,
+                    process_report.CREDIT_FIELD,
+                    process_report.SUBSIDY_FIELD,
+                    process_report.BALANCE_FIELD,
+                ]
+            ).issubset(output_df)
+        )
+
+        self.assertTrue(
+            set(["PI1", "PI2", "PI4"]).issubset(output_df["Manager (PI)"].unique())
+        )
+
+        self.assertTrue(
+            set(["ProjectA", "ProjectB", "ProjectE-test", "ProjectF"]).issubset(
+                output_df["Project"].unique()
+            )
+        )
```

Comment:
It would be nice to add assertions that PI3 isn't in the output and their projects `ProjectC-t14334` and `ProjectD` are also not in the output just to make the test complete. Because if the harvard PIs and their projects were in the output these assertions will still be good.

```
>>> projects = ["a", "b", "c", "d", "e"]
>>> set(["a", "b", "e"]).issubset(projects)
True
```